### PR TITLE
Fix usage of split_inclusive in event parser

### PR DESCRIPTION
**Pull request by cwaldren-ld**
The `test-decode-line-split-across-chunks` unit test failed locally on M1 Mac with a panic due to `unwrap()`. It was not failing in CircleCi.

I then realized our CircleCi config was using the old rust image (`circleci/rust`), so I updated to `cimg/rust`, whereupon the test began failing as it did locally. 

Due to https://github.com/rust-lang/rust/pull/89825, `split_inclusive` now returns `None` when it used to return `Some(empty slice)` in our event parser.

The version of rust in `circleci/rust` is Rust 1.49.0, which is before the issue I linked above.

### CODE REVIEW

File: eventsource-client/src/event_parser.rs [hunk 1]
```diff
@@ -281,26 +281,21 @@ impl EventParser {
...
+            if let Some(line) = lines.next() {
+                trace!(
+                    "extending line from previous chunk: {:?}+{:?}",
+                    logify(incomplete_line),
+                    logify(line)
+                );
 
-            self.last_char_was_cr = false;
-            if !line.is_empty() {
+                self.last_char_was_cr = false;
```
Comment (cwaldren-ld) [inline eventsource-client/src/event_parser.rs:298]: No longer need to check `is_empty()`.

Comment (kinyoklion) [reply eventsource-client/src/event_parser.rs:298]: Does the specification of the method allow for the possibility of the line being empty in the future?

Comment (cwaldren-ld) [reply eventsource-client/src/event_parser.rs:298]: I find the wording of the method docs [somewhat confusing to parse,](https://doc.rust-lang.org/nightly/core/primitive.slice.html#method.split_inclusive) but as far as I can tell, no. 

> Returns an iterator over subslices separated by elements that match pred. The matched element is contained in the end of the previous subslice as a terminator.

If we have `Some(match)`, that match must contain the element we were splitting around, implying the length of the match is > 0. 

> If the last element of the slice is matched, that element will be considered the terminator of the preceding slice. That slice will be the last item returned by the iterator.

If we have input `\n`, and we are splitting around `\n`, then we'll receive Some(`\n`). 

If there is no match at all, the whole slice appears to be returned. I will re-read the specification again on Monday with fresh eyes.

Comment (cwaldren-ld) [reply eventsource-client/src/event_parser.rs:303]: I added an assertion. Not sure how we feel about that. I don't want to crash customer code, but this is a guarantee that I'm relying on here. 

I also added a property test to exercise the parsing logic a bit more, with random amounts of newlines / carriage returns.

**Comment by kinyoklion** [comment]
Final question on this. Are we enforcing a min-version that has the new behavior versus the old?

File: eventsource-client/src/event_parser.rs (hunk 1 above)
Comment (cwaldren-ld) [reply eventsource-client/src/event_parser.rs:298]: Added back `is_empty()` check.

**Review by kinyoklion — APPROVED** [review]

**Pull request merged**